plugin/src/main/java/org/gradle/testretry/internal/executer/RetryTestResultProcessor.java

```diff
@@ -108,7 +111,12 @@ public void completed(Object testId, TestCompleteEvent testCompleteEvent) {
 
                 boolean failedInPreviousRound = previousRoundFailedTests.remove(className, name);
                 if (failedInPreviousRound && testCompleteEvent.getResultType() == SKIPPED) {
-                    addRetry(descriptor);
+                    if (failOnSkippedAfterRetry) {
+                        addRetry(descriptor);
+                    } else {
+                        emitFakeSkippedEvent(descriptor, testCompleteEvent, name);
```

Comment:
I don't think you need this `else`, because Gradle should report the skipped execution. I think this will cause the method to be reported twice (for example, in the console log).